languages/wikipron/src/generate_summary.py

```diff
@@ -74,38 +76,28 @@ def main() -> None:
     languages_summary_list.sort(key=_wiki_name_and_transcription_level)
     readme_list.sort(key=_wiki_name_and_transcription_level)
     # Writes the TSV.
-    with open(LANGUAGES_SUMMARY_PATH, "w") as source:
+    with open(LANGUAGES_SUMMARY_PATH, "w") as sink:
         tsv_writer_object = csv.writer(
-            source, delimiter="\t", lineterminator="\n"
+            sink, delimiter="\t", lineterminator="\n"
         )
         tsv_writer_object.writerows(languages_summary_list)
     # Writes the README.
-    with open(README_PATH, "w") as source:
-        headers = [
-            [
-                "Link",
-                "ISO 639-2 Code",
-                "ISO 639 Language Name",
-                "Wiktionary Language Name",
-                "Case-folding",
-                "Phonetic/Phonemic",
-                "# of entries",
-            ],
-            [
-                ":----",
-                ":----:",
-                ":----:",
-                ":----:",
-                ":----:",
-                ":----:",
-                "----:",
-            ],
-        ]
-        readme_list = headers + readme_list
-        formatted_and_converted_to_strings = [
-            "| " + " | ".join(ele) + " |\n" for ele in readme_list
-        ]
-        source.writelines(formatted_and_converted_to_strings)
+    with open(README_PATH, "w") as sink:
+        print(
+            "| Link | ISO 639-2 Code | ISO 639 Language Name "
+            "| Wiktionary Language Name | Case-folding "
+            "| Phonetic/Phonemic | # of entries |",
+            file=sink,
+        )
+        print(
+            "| :---- | :----: | :----: | :----: | :----: | :----: | ----: |",
+            file=sink,
+        )
+        for link, code, n1, n2, cf, ph, count in readme_list:
+            print(
+                f"| {link} | {code} | {n1} | {n2} | {cf} | {ph} | {count:,} |",
+                file=sink,
+            )
```

Comment:
This is certainly a cleaner way of doing it, my only complaint would be `n1, n2, cf, ph` are not particularly friendly/informative variable names.

Reply:
Sure, but I think this a rare case where it doesn't matter and giving them informative ones would blow up the line-wrapping so badly that the net cognitive effect would be negative.